packages/loader/container-loader/src/test/serializedStateManager.spec.ts

```diff
@@ -206,55 +237,309 @@ describe("serializedStateManager", () => {
+		const storageAdapter = new MockStorageAdapter();
+		const deferred = new Deferred<void>();
+		const cb = () => {
+			deferred.resolve();
+		};
+		const serializedStateManager = new SerializedStateManager(
+			pending,
+			logger.toTelemetryLogger(),
+			storageAdapter,
+			true,
+			cb,
+		);
+
+		const processedOpsSize = 20;
+		for (let num = 0; num < processedOpsSize; ++num) {
+			serializedStateManager.addProcessedOp(generateSavedOp());
+		}
+
+		const snapshotSequenceNumber = 11;
+		storageAdapter.uploadSummary(snapshotSequenceNumber);
+
+		await serializedStateManager.fetchSnapshot(undefined, false);
+		await deferred.promise;
+		const state = await serializedStateManager.getPendingLocalStateCore(
+			{ notifyImminentClosure: false },
+			"clientId",
+			new MockRuntime(),
+			resolvedUrl,
+		);
+		const parsed = JSON.parse(state) as IPendingContainerState;
+		assert.strictEqual(parsed.baseSnapshot.id, "fromStorage");
+		const attributes = getAttributesFromPendingState(parsed);
+		assert.strictEqual(
+			attributes.sequenceNumber,
+			snapshotSequenceNumber,
+			"wrong snapshot sequence number",
+		);
+		assert.strictEqual(
+			parsed.savedOps[0].sequenceNumber,
+			snapshotSequenceNumber + 1,
+			"wrong first saved op",
+		);
+		assert.strictEqual(
+			parsed.savedOps[parsed.savedOps.length - 1].sequenceNumber,
+			processedOpsSize,
+			"wrong last saved op",
+		);
+	});
+
+	it("refresh snapshot when there are no processed ops", async () => {
+		const pending: IPendingContainerState = {
+			...pendingLocalState,
+			baseSnapshot: { ...snapshot, id: "fromPending" },
+		};
+		const storageAdapter = new MockStorageAdapter();
+		const deferred = new Deferred<void>();
+		const cb = () => {
+			deferred.resolve();
+		};
+		const serializedStateManager = new SerializedStateManager(
+			pending,
+			logger.toTelemetryLogger(),
+			storageAdapter,
+			true,
+			cb,
+		);
+		const snapshotSequenceNumber = 11;
+		storageAdapter.uploadSummary(snapshotSequenceNumber);
+
+		await serializedStateManager.fetchSnapshot(undefined, false);
+		await deferred.promise;
+		const state = await serializedStateManager.getPendingLocalStateCore(
+			{ notifyImminentClosure: false },
+			"clientId",
+			new MockRuntime(),
+			resolvedUrl,
+		);
+		const parsed = JSON.parse(state) as IPendingContainerState;
+		assert.strictEqual(parsed.baseSnapshot.id, "fromStorage");
+		const attributes = getAttributesFromPendingState(parsed);
+		assert.strictEqual(
+			attributes.sequenceNumber,
+			snapshotSequenceNumber,
+			"wrong snapshot sequence number",
+		);
+		assert.strictEqual(parsed.savedOps.length, 0, "should be no savedOps");
+	});
+
+	it("refresh snapshot when snapshot sequence number is above processed ops", async () => {
+		const pending: IPendingContainerState = {
+			...pendingLocalState,
+			baseSnapshot: { ...snapshot, id: "fromPending" },
+		};
+		const storageAdapter = new MockStorageAdapter();
+		const deferred = new Deferred<void>();
+		const cb = () => {
```

Comment:
can you improve the names here. its hard to follow the test as the current names don't give any hint of what they will be used for